### PR TITLE
chore: simplify test setup and remove unused deps

### PR DESCRIPTION
**Pull request by jo**
This simplifies the test setup by using only tap for coverage. Also remove
unused dev dependencies.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,12 +4,9 @@
   "description": "Central place to manage kazana entities",
   "main": "index.js",
   "scripts": {
-    "coverage": "nyc report --reporter=text-lcov | coveralls",
-    "test": "nyc tap tests/**/*-test.js",
-    "test:unit": "tap tests/unit/*-test.js | tap-spec",
+    "test": "tap --coverage tests/**/*-test.js",
     "start": "kazana --bare",
-    "semantic-release": "semantic-release pre && npm publish && semantic-release post",
-    "postsemantic-release": "npm run coverage"
```
Comment (gr2m) [inline package.json]: Don’t we still need to set it up to send coverage to coveralls?

Comment (jo) [inline package.json]: This is all done by tap

Comment (jo) [inline package.json]: see https://coveralls.io/builds/4662597 \o/

Comment (gr2m) [inline package.json]: Oh wow, interesting

Comment (jo) [inline package.json]: https://github.com/tapjs/node-tap#travis-ci-and-coverallsiocodecovio-integration

Comment (jo) [inline package.json]: Also related: https://github.com/tapjs/node-tap/issues/210

**Comment by quesebifurcan** [comment]
+1

**Pull request merged**